lightning/src/chain/chaininterface.rs

```diff
@@ -71,22 +71,10 @@ pub trait BroadcasterInterface: Sync + Send {
 
 /// A trait indicating a desire to listen for events from the chain
 pub trait ChainListener: Sync + Send {
-	/// Notifies a listener that a block was connected.
-	///
-	/// The txn_matched array should be set to references to transactions which matched the
-	/// relevant installed watch outpoints/txn, or the full set of transactions in the block.
-	///
-	/// Note that if txn_matched includes only matched transactions, and a new
-	/// transaction/outpoint is watched during a block_connected call, the block *must* be
-	/// re-scanned with the new transaction/outpoints and block_connected should be called
-	/// again with the same header and (at least) the new transactions.
-	///
-	/// Note that if non-new transaction/outpoints are be registered during a call, a second call
-	/// *must not* happen.
-	///
-	/// This also means those counting confirmations using block_connected callbacks should watch
-	/// for duplicate headers and not count them towards confirmations!
-	fn block_connected(&self, header: &BlockHeader, height: u32, txn_matched: &[&Transaction], indexes_of_txn_matched: &[usize]);
+	/// Notifies a listener that a block was connected. Transactions may be filtered and are given
+	/// paired with their position within the block.
+	fn block_connected(&self, header: &BlockHeader, txdata: &[(usize, &Transaction)], height: u32);
```

Comment:
Can you elaborate a bit more on the reasoning for the two-slices -> slice-of-pairs change? I think I agree its more of the "correct" API, but dunno if its worth the change which will probably require some work on the bindings end - did you have a specific reasoning in mind or just think its cleaner?

Reply:
Partly to make it cleaner and consistent throughout. Also less error prone since listeners don't need to manually match up transactions and indexes. Ideally, the indexes are not needed. Right now it seems specific to `ChannelManager`, but perhaps there are other uses that I haven't thought of.